pkg/kv/kvserver/store_raft.go

```diff
@@ -303,7 +303,7 @@ func (s *Store) HandleRaftUncoalescedRequest(
 	// count them.
 	s.metrics.RaftRcvdMessages[req.Message.Type].Inc(1)
 
-	q, _ := s.raftRecvQueues.LoadOrCreate(req.RangeID)
+	q, _ := s.raftRecvQueues.LoadOrCreate(req.RangeID, s.cfg.RaftMaxInflightMsgs)
```

Comment:
We should maybe add some room for additional messages like heartbeats, like @nvanbenschoten mentioned over on the issue. Something like 10 is probably sufficient. Can we see what happens if the receive queue fills up -- will the sender still send messages like heartbeats?

Was also thinking we should treat 0 as unlimited here, but looks like the Raft lib explicitly disallows 0:

https://github.com/cockroachdb/vendored/blob/4d6e89235f6ac5c478ebcc3e9ef073914ac7348e/go.etcd.io/etcd/raft/v3/raft.go#L228-L230

Reply:
I'm still reading on `RaftMaxInflightMsgs`, but it seems like it controls the window size of the in-flight entries, and not necessarily the number of outgoing messages (because batching etc), so I wonder whether we are comparing apples to apples here at all.

Reply:
>What are other types of messages that we should "prioritize"?

Heartbeats would be the main one.

>the receiver could filter based on the message type, e.g., always let heartbeats in, or have a separate limit for them (and for other important messages)

Because these queues are FIFO, we wouldn't want to unconditionally allow e.g. heartbeats in, because they could (in principle) use an unbounded amount of memory if there's any head-of-line blocking.

We could change this to be a priority queue or some such, but that's a larger lift, and certainly won't be backportable.

For now, let's just do something quick and simple that we can merge today. `RaftMaxInflightMsgs` + 10 seems like a reasonable improvement over the current state.

>I wonder if, to reduce the coupling, we should introduce a second config setting instead, which might sync with RaftMaxInflightMsgs by default (and have some custom +10 thing, as well as validation).

I think coupling is fine here -- it's one less thing for operators to get wrong, and one less thing for TSEs to check during escalations.

Reply:
>I'm still reading on RaftMaxInflightMsgs, but it seems like it controls the window size of the in-flight entries, and not necessarily the number of outgoing messages (because batching etc), so I wonder whether we are comparing apples to apples here at all.

Hm, that doesn't seem to match the documentation at least. Reference?

Reply:
Nah, `MaxInflightMsgs` does track messages, not entries. However, it contains the index of the last entry in the message. See:

https://github.com/etcd-io/etcd/blob/682851796524be86f958d1526ad579938b5cdd50/raft/tracker/progress.go#L64-L76

https://github.com/etcd-io/etcd/blob/87258efd90224bc8b59e000f75fe07fdeab68e2d/raft/raft.go#L474-L476

Reply:
Ah right, messages indeed. I'm also checking how that works with "coalesced" heartbeat messages.

Reply:
From Raft's point of view, I think they're just regular messages -- we do the coalescing at a layer above.

Reply:
Yep, each heartbeat is counted as 1 message after uncoalescing. That seems fine.

Added +10, as suggested.